reference-architectures/cloud_deploy_flow/main.tf

```diff
@@ -0,0 +1,176 @@
+# Configure the Google Cloud provider
+terraform {
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.0"
+    }
+  }
+  provider_meta "google" {
+    module_name = "cloud-solutions/platform-engineering-cloud-deploy-pipeline-deploy-v1"
+  }
+}
+
+variable "gcp_service_list" {
+  description = "List of GCP Services to enable (WIP)"
+  type = list(string)
+  default = [
+    "pubsub.googleapis.com",
+    "clouddeploy.googleapis.com",
+    "cloudbuild.googleapis.com"
+  ]
+}
+
+# Enable Services (Work in Progress)
+resource "google_project_service" "project" {
+  for_each = toset(var.gcp_service_list)
+  project = var.project_id
+  service = each.key
+
+  timeouts {
+    create = "30m"
+    update = "40m"
+  }
+
+  disable_on_destroy = false
+}
+
+# Create a Pub/Sub topic for commands sent to CF to interact with Cloud Deploy
+resource "google_pubsub_topic" "deploy-commands" {
+  name = "deploy-commands"
+  project = var.project_id
+}
+
+# Create a Pub/Sub subscription for deploy-commands topic
+resource "google_pubsub_subscription" "deploy_commands_subscription" {
+  name  = "deploy-commands-subscription"
+  topic = google_pubsub_topic.deploy-commands.id
+  project = var.project_id
+}
+
+# Create a Pub/Sub topic to receive Cloud Deploy Operations Notifications
+resource "google_pubsub_topic" "deploy_operations" {
+  name = "clouddeploy-operations"
+  project = var.project_id
+}
+
+# Create a Pub/Sub subscription for clouddeploy-operations topic
+resource "google_pubsub_subscription" "deploy_operations_subscription" {
+  name  = "clouddeploy-operations-subscription"
+  topic = google_pubsub_topic.deploy_operations.id
+  project = var.project_id
+}
+
+resource "google_pubsub_topic" "deploy_approvals" {
+  name = "clouddeploy-approvals"
+  project = var.project_id
+}
+
+# Create a Pub/Sub subscription for clouddeploy-approvals topic
+resource "google_pubsub_subscription" "deploy_approvals_subscription" {
+  name  = "clouddeploy-approvals-subscription"
+  topic = google_pubsub_topic.deploy_approvals.id
+  project = var.project_id
+}
+
+# Create a Pub/Sub topic to receive Cloud Build Notifications
+resource "google_pubsub_topic" "build_notifications" {
+  name = "cloud-builds"
+  project = var.project_id
+}
+
+# Create a Pub/Sub subscription for clouddeploy-approvals topic
+resource "google_pubsub_subscription" "build_notifications_subscription" {
+  name  = "build_notifications_subscription"
+  topic = google_pubsub_topic.build_notifications.id
+  project = var.project_id
+}
```

Comment:
Consider using `for_each` to make this more concise.

Reply:
Good call out, didn't think I could and still be able to reference them but found a solution :)